driver-core/src/main/com/mongodb/internal/connection/MongoCredentialWithCache.java

```diff
@@ -63,28 +63,23 @@ public void putInCache(final Object key, final Object value) {
         cache.set(key, value);
     }
 
-    public Lock getLock() {
-        return cache.lock;
+    public <V> V withLock(final Supplier<V> k) {
+        return Locks.withLock(cache.lock, k);
     }
 
     static class Cache {
         private final ReentrantLock lock = new ReentrantLock();
-        private Object cacheKey;
-        private Object cacheValue;
+        private final ConcurrentHashMap<Object, Object> cache = new ConcurrentHashMap<>();
```

Comment:
Is a requirement for support of multiple cache keys coming?  Is that the motivation for this change?

Reply:
I added an `OidcCacheEntry`, to be stored under its own key.

It seemed that if we support multiple cache keys at all (the implication is that we do, because we have a "key" field), then they should be mapped separately (as if under different namespaces), and it was an oversight to do otherwise. This is the safest way to avoid interfering with (and needing to think about) other entries.

Though, looking at the usages, it appears that we would not share MongoCredentialWithCache across authenticators. Also, the usage in `ScramShaAuthenticator` potentially depends on having one "key" at a time, but that logic seems specific to that authenticator and shouldn't be part of the cache.

Reply:
Reverted and documented.